### DISCUSSION TITLE
LoRa project information

### DISCUSSION THREAD

**Discussion by davefes** [micropython/micropython#13233] [Show and tell]
Came across this blog and thought it that this guy seems to be a good source of LoRa information.
https://stuartsprojects.github.io

Especially, some articles on reliable comms.
https://stuartsprojects.github.io/2021/08/25/Reliable-Packets-Part1.html
https://stuartsprojects.github.io/2021/08/26/Reliable-Packets-Part2.html

**Comment by mcauser** [comment]
With point-to-point LoRa (and other modulations) you need to worry about reliability with transmissions, otherwise it's a bit fire-and-forget.

LoRaWAN has a concept of Confirmed Uplinks, where it can be configured to repeatedly attempt sending the same payload until it receives a confirmation (downlink). Could pinch some ideas from here when building custom P2P solutions.

That blog's an interesting read, thanks for sharing

**Comment by kjm1102** [comment]
The phantom bytes thing is new, haven't seen that before.

Re data integrity I like to use internal consistency validation as a last line of defence. eg check that analog quantities are within range, that redundant copies of on/off control bits all agree, that a new flow pulse total is not less than the previous value, etc.

**Comment by sophiedegran** [comment]
Thanks dave very nice Site and travel with balloon! ☺

**Comment by peterhinch** [comment]
Some interesting information there. One thing that astonished me was the point that a single bit error would cause the RX to ignore the checksum. On the face of it that's a real rookie error in the LoRa protocol design. Unless you can program the RX to require a checksum regardless of the state of the bit?

The NRF24L01 solves several of these problems in silicon, but of course the range is short.

**Comment by zcattacz** [comment]
This reminds me of phantom EPCs from UHF RFID tags. 
I didn't notice it until the reader is installed on site to read tags on hi-speed spinning metal surface. The chance is low but not rare enough to be ignored. It gives all sorts of malformed EPC data: last byte altered, extra trailing bytes, or complete random. All of the logged have correct matching PC bytes to be verified. Initially I thought I could just use the EPC coming with the tags, but in this case a stronger CRC and extra padding is definitely necessary.

**Comment by jken12** [comment]
Thank you so much Davy[,](https://smartglasspro.ae/) Site is amazing
